### ISSUE TITLE
Unable to access/remove empty encrypted files

### ISSUE THREAD

**Issue by kvinwang** [gramineproject/gramine#1349]
### Description of the problem

We are trying to put RocksDB files in an encrypted files directory in Gramine. The problem is that our program usually failed to restart with an error "`Permission denied`" when it was terminated by `Ctrl-C` or a power cut-off. 
The reason for the error is that when the program is force terminated, there might be some empty files left in the database directory, such as the `LOCK`, and `LOG`:
```
drwxr-xr-x 2 kvin kvin   4096 5月  16 09:51 ./
drwxr-xr-x 3 kvin kvin   4096 5月  16 09:51 ../
-rw-r--r-- 1 kvin kvin 401408 5月  16 09:51 000004.log
-rw-r--r-- 1 kvin kvin   4096 5月  16 09:51 CURRENT
-rw-r--r-- 1 kvin kvin   4096 5月  16 09:51 IDENTITY
-rw-r--r-- 1 kvin kvin      0 5月  16 09:51 LOCK
-rw-r--r-- 1 kvin kvin      0 5月  16 09:51 LOG
-rw-r--r-- 1 kvin kvin   4096 5月  16 09:51 MANIFEST-000005
-rw-r--r-- 1 kvin kvin  12288 5月  16 09:51 OPTIONS-000007
-rw-r--r-- 1 kvin kvin  12288 5月  16 09:51 OPTIONS-000009
```
Then after restarting, those empty files no longer to be able to be accessed by the program, with the following errors:
```
(libos_fs_encrypted.c:141:cb_debug) [P1:T1:pruntime] debug: ipf_open: handle: 0, path: 'data/protected_files/caches/cache_0/LOG', real size: 0, mode: 0x3
(libos_fs_encrypted.c:41:cb_read) [P1:T1:pruntime] warning: EOF
(libos_fs_encrypted.c:141:cb_debug) [P1:T1:pruntime] debug: ipf_open: failed: -11
(libos_fs_encrypted.c:203:encrypted_file_internal_open) [P1:T1:pruntime] warning: pf_open failed: Callback failed
(libos_fs_encrypted.c:141:cb_debug) [P1:T1:pruntime] debug: ipf_open: handle: 0, path: 'data/protected_files/caches/cache_0/LOCK', real size: 0, mode: 0x3
(libos_fs_encrypted.c:41:cb_read) [P1:T1:pruntime] warning: EOF
(libos_fs_encrypted.c:141:cb_debug) [P1:T1:pruntime] debug: ipf_open: failed: -11
(libos_fs_encrypted.c:203:encrypted_file_internal_open) [P1:T1:pruntime] warning: pf_open failed: Callback failed
```

**I've considered `removing the entire database directory and recreating it` as a second choice, but it doesn't work either, the empty files can not be removed in Gramine either.**

### Steps to reproduce

Mount `/protected_files/` as encrypted, then compile and run the following workload:
```C
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>

int main() {
    const char* filename = "/protected_files/hello";
    if (access(filename, F_OK) != -1) {
        if (remove(filename) != 0) {
            perror("Error removing file");
            return 1;
        }
    }

    FILE *file = fopen(filename, "w");
    if (file == NULL) {
        perror("Error opening file");
        return 1;
    }
    printf("Press Control-C to exit\n");
    fflush(stdout);
    while(1) {
        sleep(1);
    }
}
```

Press Ctrl-C to terminate it and run again.

### Expected results

It should be able to run the second time.

### Actual results

It would fail to restart.

### Gramine commit hash

aefd5d0875c4fdf76edd67f5043952bdd32b0f7a

**Comment by kvinwang** [comment]
Is it a choice to add an option to the manifest like:

```
fs.mounts = [
  { type = "encrypted", path = "[PATH]", uri = "[URI]", key_name = "[KEY_NAME]", allow_empty_files = true },
]
```

**Comment by dimakuv** [comment]
@kvinwang Could you try the latest Gramine master? I fixed a similar problem ~2 months ago: https://github.com/gramineproject/gramine/commit/032e35ecf32096cc3e8134f90dadad770b545fe0

I think this fix should also work in your case. The fix basically flushes the encrypted file immediately on creation/open, which allows to populate the underlying host file with correct metadata immediately, and Gramine won't complain later.

**Comment by kvinwang** [comment]
> @kvinwang Could you try the latest Gramine master? I fixed a similar problem ~2 months ago: [032e35e](https://github.com/gramineproject/gramine/commit/032e35ecf32096cc3e8134f90dadad770b545fe0)
> 
> I think this fix should also work in your case. The fix basically flushes the encrypted file immediately on creation/open, which allows to populate the underlying host file with correct metadata immediately, and Gramine won't complain later.

It works with the latest master. Thanks.
